platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -174,3 +174,67 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
+
+=== Explanation
+
+The preferred location for build logic is an included build named `build-logic`, **not** in `buildSrc`.
+
+The automatically available `buildSrc` is great for rapid prototyping, but it comes with some subtle disadvantages:
+- There are classloader differences in how these 2 approaches behave that can be surprising; included builds are treated just like external dependencies, which is a simpler mental model.  Dependency resolution behaves subtly differently in buildSrc.
+- There can potentially be fewer task invalidations in a build when files in an included build are modified, leading to faster builds.
+- Included builds are complete Gradle builds and can be opened, worked on, and built independently as standalone projects.  It is straightforward to publish their products, including plugins, in order to share them with other projects.
+
+Ideally, there would be no difference between using `buildSrc` and an included build, `buildSrc` is intended to behave like an implicitly available included build.
+However, due to historical reasons, some subtle differences still exist.
+Gradle is working to unify the behavior of these two approaches, and this recommendation may be revised in the future.
+For now, these differences can introduce confusion and are best avoided.
+
+Since setting up a composite build requires only minimal additional configuration, we recommend using it over `buildSrc`.
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-avoid/kotlin/buildSrc",files="build.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-avoid/groovy/buildSrc",files="build.gradle[tags=avoid-this]"]
+====
+
+*Setup a Plugin Build*: This is the same using either method.
+
+====
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-avoid/kotlin",files="settings.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-avoid/groovy",files="settings.gradle[tags=avoid-this]"]
+====
+
+*BuildSrc products are automatically usable*: There is no additional configuration with this method.
+
+==== Do This Instead
+
+====
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-do/kotlin/build-logic",files="build.gradle.kts[tags=do-this]"]
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-do/groovy/build-logic",files="build.gradle[tags=do-this]"]
+====
+
+*Setup a Plugin Build*: This is the same using either method.
+
+====
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-do/kotlin",files="settings.gradle.kts[tags=do-this]"]
+include::sample[dir="snippets/bestPractices/favorCompositeBuilds-do/groovy",files="settings.gradle[tags=do-this]"]
```

Comment:
Nice to have comment - I would name both projects the same and remove the "avoid"/"do" part. This is easy to understand when you see both, but when I've seen only the "do" one it took me a second to understand this.

Reply:
The directories need to be distinct, but I can update the `rootProject.name` in the settings files.  Is this what you mean?

Reply:
Yeah, it's about what user sees in the snippets :)